tests/chart_tests/test_houston_configmap.py

```diff
@@ -232,6 +232,42 @@ def test_houston_configmap_with_loggingsidecar_enabled_with_overrides():
     assert "vector" in prod_yaml["deployments"]["loggingSidecar"]["image"]
 
 
+def test_houston_configmap_with_loggingsidecar_enabled_with_indexPattern():
+    """Validate the houston configmap and its embedded data with
+    loggingSidecar."""
+    sidecar_container_name = "sidecar-log-test"
+    image_name = "example.com/some-repo/test-image-name:test-tag-foo"
+    indexPattern = "%Y.%m"
```

Comment:
This parametrize block works and shows that the single-quote val breaks the helm mechanic.
```suggestion
@pytest.mark.parametrize(
    "indexPattern",
    [
        "%Y.%m",
        "%Y.%m.%d",
        "a-single-quote-'-in-the-indexPattern",
        'a-double-quote-"-in-the-indexPattern',
    ],
)
def test_houston_configmap_with_loggingsidecar_enabled_with_indexPattern():
    """Validate the houston configmap and its embedded data with
    loggingSidecar."""
    sidecar_container_name = "sidecar-log-test"
    image_name = "example.com/some-repo/test-image-name:test-tag-foo"
```

Reply:
Another way we could work around this is to add a comment that single-quotes are not allowed. That would be a really easy non-code way to handle this potential problem.

Reply:
I tried to find a way to handle both single and double quotes and I don't see an easy way to do it without making really ugly code. Helm printf does not appear to have any mechanism to produce the right string. I think we can skip merging the parametrize change and just put a comment in the values.yaml that quote characters are not supported in the indexPattern.